statement_parser_test.go

```diff
@@ -465,61 +466,186 @@ SELECT SchoolID FROM Roster`,
 
 func TestFindParams(t *testing.T) {
 	tests := []struct {
-		input   string
-		want    []string
-		wantErr bool
+		input              string
+		wantSQL            string
+		want               []string
+		wantErr            error
+		skipRemoveComments bool
 	}{
 		{
-			input: `SELECT * FROM PersonsTable WHERE id=@id`,
-			want:  []string{"id"},
+			input:   `SELECT * FROM PersonsTable WHERE id=@id`,
+			wantSQL: `SELECT * FROM PersonsTable WHERE id=@id`,
+			want:    []string{"id"},
 		},
 		{
-			input: `SELECT * FROM PersonsTable WHERE id=@id AND name=@name`,
-			want:  []string{"id", "name"},
+			input:   `SELECT * FROM PersonsTable WHERE id=@id AND name=@name`,
+			wantSQL: `SELECT * FROM PersonsTable WHERE id=@id AND name=@name`,
+			want:    []string{"id", "name"},
 		},
 		{
-			input: `SELECT * FROM PersonsTable WHERE Name like @name AND Email='test@test.com'`,
-			want:  []string{"name"},
+			input:   `SELECT * FROM PersonsTable WHERE Name like @name AND Email='test@test.com'`,
+			wantSQL: `SELECT * FROM PersonsTable WHERE Name like @name AND Email='test@test.com'`,
+			want:    []string{"name"},
 		},
 		{
 			input: `SELECT * FROM """strange
- @table
-""" WHERE Name like @name AND Email='test@test.com'`,
+		@table
+		""" WHERE Name like @name AND Email='test@test.com'`,
+			wantSQL: `SELECT * FROM """strange
+		@table
+		""" WHERE Name like @name AND Email='test@test.com'`,
```

Comment:
Would you mind adding a test that has a triple-quoted identifier as the last part of the SQL string. So for example

```sql
select * from foo where ?=```strange @table```
```

Reply:
Added the test, behaviour is similar to what JDBC spanner will return.